core/cont/src/TList.cxx

```diff
@@ -713,30 +716,31 @@ void TList::RecursiveRemove(TObject *obj)
    TObjLink *next = 0;
    while (lnk) {
       next = lnk->Next();
-      TObject *ob = lnk->GetObject();
-      if (ob->TestBit(kNotDeleted)) {
-         if (ob->IsEqual(obj)) {
-            if (lnk == fFirst) {
-               fFirst = next;
-               if (lnk == fLast)
-                  fLast = fFirst;
-               else
-                  fFirst->fPrev = 0;
-               DeleteLink(lnk);
-            } else if (lnk == fLast) {
-               fLast = lnk->Prev();
-               fLast->fNext = 0;
-               DeleteLink(lnk);
-            } else {
-               lnk->Prev()->fNext = next;
-               lnk->Next()->fPrev = lnk->Prev();
-               DeleteLink(lnk);
-            }
-            fSize--;
-            fCache = 0;
-            Changed();
-         } else
-            ob->RecursiveRemove(obj);
+      if (TObject *ob = lnk->GetObject()) {
```

Comment:
Do you think this can be converted into a `for` loop, too? Maybe even with
```
if (!ob->TestBit(kNotDeleted))
   continue;
if (!ob->IsEqual(obj)) {
   ob->RecursiveRemove(obj);
   continue;
}
```
to reduce brain stack and when reading this code, and indentation?

Reply:
Sure, I will try.